File: CSharp/5kyu/WeightForWeight.cs
using System.Linq;
using System;
using System.Collections.Generic;

namespace CSharp._5kyu
{
    public static partial class Kata5
    {
        public class WeightComparer : IComparer<string>
        {
            public int Compare(string x, string y)
            {
                return x.Weight() == y.Weight() ? string.Compare(x, y) : x.Weight() - y.Weight();
            }
        }
        public static string OrderWeight(string strng)
        {
            var result = strng.Split(" ", StringSplitOptions.RemoveEmptyEntries).OrderBy(a => a, new WeightComparer());

            return String.Join(" ", result);
        }

        public static int Weight(this string w) => w.ToCharArray().Sum(i => int.Parse(i.ToString()));

        public static string OrderWeight_Kata_1(string s)
        {
            return String.Join(" ", s.Split(" ")
                .OrderBy(c => c.ToCharArray().Select(a => (int)char.GetNumericValue(a)).Sum())
                .ThenBy(n => n));
        }
    }
}

File: JavaScript/5kyu/WeightForWeight.js
function orderWeight(strng) {
  return strng
    .split(" ")
    .sort((a, b) => {
      if (a.weight() == b.weight()) {
        return a > b ? 1 : -1;
      }

      return a.weight() - b.weight();
    })
    .join(" ");
}

function orderWeight_kata_1(strng) {
  return strng
    .split(" ")
    .sort((a, b) =>
      a.weight() == b.weight() ? a.localeCompare(b) : a.weight() - b.weight()
    )
    .join(" ");
}

String.prototype.weight = function () {
  return this.split("").reduce((p, c) => p + parseInt(c), 0);
};

var result = orderWeight_kata_1(
  "2000 10003 1234000 44444444 9999 11 11 22 123"
);
console.log("11 11 2000 10003 22 123 1234000 44444444 9999");
console.log(result);
